lib/services/plugins.test.js

```diff
@@ -43,23 +44,59 @@ describe(_.startCase(filename), function () {
     const fn = lib[this.title];
 
     it('will try to invoke each plugin', function () {
-      lib.plugins = [plugin];
+      lib.plugins = [plugin, pluginWithMetadata];
       plugin.returns(true);
 
       return fn()
         .then(() => {
           sinon.assert.calledOnce(plugin);
+          sinon.assert.calledOnce(pluginWithMetadata.init);
         });
     });
 
-    it('logs a warning if a plugin is not a function', function () {
+    it('logs a warning if a plugin is not a function or does not have an init function', function () {
       lib.plugins = [false];
       plugin.returns(true);
 
       return fn()
         .then(() => {
-          sinon.assert.calledWith(fakeLog, 'warn', 'Plugin is not a function');
+          sinon.assert.calledWith(fakeLog, 'warn', 'Plugin is not a function or does not have an init function');
         });
     });
   });
+
+  describe('getWhitelistFromPlugins', function () {
+    it('returns an array of all the plugin\'s whitelist combined', function () {
```

Comment:
Can we cache the function that's being tested so we don't have to call `lib.getWhitelistFromPlugins` in every test case?
```js
describe('getWhitelistFromPlugins', function () {
  const fn = lib[this.title];

  it('returns an array of all the plugin\'s whitelist combined', function () {
    // ...

    expect(fn(plugins)).to.eql(['/_test']);
  });
});
```

Do this for every test suite